app/src/main/java/org/openhdp/hdt/ui/dashboard/DashboardViewModel.kt

```diff
@@ -0,0 +1,142 @@
+package org.openhdp.hdt.ui.dashboard
+
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.*
+import org.openhdp.hdt.data.StopwatchRepository
+import timber.log.Timber
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+
+class DashboardViewModel @ViewModelInject constructor(
+    val stopwatchRepository: StopwatchRepository
+) : ViewModel(), OnDashboardItemClickListener {
+
+    private val _viewState = MutableLiveData<DashboardViewState>()
+
+    val viewState: LiveData<DashboardViewState>
+        get() = _viewState
+
+    private var job: Job? = null
+
+    fun midnight() = Date().apply {
+        seconds = 0
+        minutes = 0
+        hours = 0
+    }
+
+    fun initialize() {
+        _viewState.value = DashboardViewState.Results(
+            arrayListOf(
+                DashboardItem(
+                    "0",
+                    "Netflix",
+                    "",
+                    midnight().time + TimeUnit.HOURS.toMillis(2),
+                    DashboardItemState.INACTIVE
+                ),
+                DashboardItem(
+                    "1",
+                    "spacer",
+                    "",
+                    midnight().time + TimeUnit.HOURS.toMillis(1),
+                    DashboardItemState.ACTIVE
+                ),
+                DashboardItem(
+                    "2",
+                    "angielski",
+                    "",
+                    midnight().time,
+                    DashboardItemState.ACTIVE
+                )
+            )
+        )
+
+        countdown()
+    }
+
+    private fun countdown() {
```

Comment:
We could also use ticking seconds for UI representation of time, and for example elapsedRealtime() for database representation to avoid problems with interrupted ticking.